packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart

```diff
@@ -466,7 +471,11 @@ class DynamiteClient {
     if (validStatuses?.contains(response.statusCode) ?? true) {
       return response;
     } else {
-      throw await DynamiteApiException.fromResponse(response);
+      if (kDebugMode) {
+        throw await DynamiteStatusCodeException.fromResponse(response);
+      } else {
+        throw DynamiteStatusCodeException(response.statusCode);
+      }
```

Comment:
Can't you move this logic into the factory? That would make it easier and impossible to make a mistake when only using one of them

Reply:
But the method is asynchronous. So the event loop would still be interrupted for no reason.
I'll think about a better solution. Now that this is only used in debug mode we can just block the execution by synchronously waiting.

Reply:
Ah I see